src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -45,6 +49,15 @@ public void Configure(TelemetryConfiguration configuration)
                 configuration.InstrumentationKey = this.applicationInsightsServiceOptions.InstrumentationKey;
             }
 
+            if (this.telemetryProcessorFactories.Any())
```

Comment:
@karolz-ms and @pharring Would we have a "race" condition where someone could add a processor factory after we call Build()?  Something like the issue with calling both UseApplicationInsights and AddApplicationInsightsTelemetry?